dev_docs/cxg.md

```diff
@@ -0,0 +1,162 @@
+
+The `cxg_schema` JSON string is attached to the TileDB array metadata, and is a dictionary containing the following top-level names:
+* "index": string, containing the name of the index column
+* \<column-name\>: optional, a JSON dict, contain a schema definition using the same format as the cellxgene REST API /schema route
+
+For example:
+```
+{
+    "index": "obs_index",
+    "louvain": { "type": "categorical", "categories": [ "0", "1", "2", "3", "4" ]}
+    "is_useful": { "type": "boolean" }
+}
+```
+
+### X
+
+TileDB array, with a single anonymous attribute, shape (n_obs, n_var), containing the count matrix (equivalent to the AnnData `X` array).  MUST have numeric type, and SHOULD be float32.  The TileDB schema defines type and sparsity, and both dense and sparse encoding are supported.
+
+### X_col_shift
+
+Optional TileDB array, used to encode-per column offsets for column-shift sparse encoding.  The TileDB array will have a single anonymous attribute, of the same type as the X array, and shape (n_var,).  
+
+If the X array is sparse, and X_col_shift exists, then all values in the i'th column were subtracted by X_col_shift[i].
+
+### emb and embedding arrays
+
+A CXG must have a group named `emb`, which will contain all embeddings.  Embeddings are encoded as TileDB arrays, of numeric type and shape (n_obs, >=2).  The arrays SHOULD be coerced to float32, and MUST be a numeric type.  The TileDB array name will be assumed to be the embedding name (conventionally, embedding names in CXG are _not_ prefixed with an `X_` as they are in AnnData).
+
+CXG supports zero or more embeddings. Note that cellxgene currently _requires_ at least one embedding.
+
+### cxg_group_metadata
+
+Required, but empty TileDB array, used to store CXG-wide metadata.  The following fields are defined:
+* __cxg_version__: (required) a semver string identifying the specification version used to encode the CXG.
+* __cxg_properties__: (optional) a dictionary containing dataset wide properties, defined below.
+* __cxg_category_colors__: (optional) a categorical color table, defined below.
+
+#### cxg_properties
+
+The properties metadata dictionary contains dataset-wide properties, encoded as a JSON dictionary. Currently, the following fields are defined:
+* title: string, dataset human name (eg, "Lung Tissue")
+* about: string, fully-qualified http/https URL, linking to more inforation on the dataset.
+
+All implementions MUST ignore unrecognized fields. 
+
+#### cxg_category_colors
+
+This optional field contains a copy of the category color table, which MAY be used to display category-specific color labels. This is a JSON dictionary, containing a per-category color-table.  Each color table is named `{category_name}_colors`, and is itself a dictionary mapping label name to RGB color.  For example:
+
+```
+{
+    "louvain_colors": {
+        "0": "#FFFFFF",
```

Comment:
It currently supports several formats, but that is really an app convention (currently unspecified), and I'm not sure if it belongs in this spec or not.  At the moment, it will support either of:
* rgb(DDD,DDD,DDD) -- eg, `rgb(255,255,255)` (base 10)
* #HHHHHH -- eg, `#FF0000` (hex)

Reply:
Okay thanks, was mostly just curious.